src/interface/sparse_pack_base.cpp

```diff
@@ -165,6 +165,10 @@ SparsePackBase SparsePackBase::Build(T *pmd, const PackDescriptor &desc,
   pack.bounds_ = bounds_t("bounds", 2, nblocks, nvar + 1);
   pack.bounds_h_ = Kokkos::create_mirror_view(pack.bounds_);
 
+  // This array stores refinement levels of current block and all neighboring blocks.
+  pack.block_props_ = block_props_t("block_props", nblocks, 2, 3, 3, 3);
```

Comment:
Having a `1,3,3,3` block of memory just to store the current block's `gid` is wasteful but we would need to go to a `4,4,4` matrix to store all possible neighbor `gid`s and I don't know that that information is needed anywhere currently.

Reply:
One option would be to make this a `ParArray2D<int>` with shape `(nblock, 27 + 1)` array and flatten the index space in the `GetLevel` calls. 

If it was flattened, another option would be to fold this into `bounds_` by changing its shape to `(2, nblocks, nvar + 1 + 27)`. This would eliminate the added `deep_copy` but keeps the same amount of extra storage as you have now.

Reply:
At least as the `gid`s are assigned currently, I think we could store one of the neighbor block gids and recover the rest since they should be in z-order within  the coarse parent block. Not sure this matters as you point out.

Reply:
Thanks for adding the `gid` to the sparse pack @brryan! 😸 Now that we are storing block refinement levels in packs, it seems only natural to also store the `gid`. 

This MR should then supersede #1010.  One of the original arguments in favor of using `VariableState` for `gid` was to avoid added bloat to the `SparsePack`.  To accommodate the `gid` here in SparsePacks, it seems that we are being a bit superfluous with the ParArray memory.... what do you think of @lroberts36 idea to shift to a flattened space?

Reply:
I followed @lroberts36 suggestion to make this a `ParArray2D`. This reduces memory footprint but not number of deep copies. I decided the code would get too ugly with smooshing bounds and gid and block properties together. Maybe we can revisit in the future especially if we keep adding data to the sparse pack. 

Still only storing the `gid` of the current meshblock, not neighbors of the current meshblock.

Reply:
Yeah, I don't have a sense for whether or not these deep copies matter. My guess is they are only likely to have any impact in codes that employ sparse fields (since the packs are likely to be rebuilt more often there). I agree that it is cleaner to separate them out for now.

Reply:
Good point about this only being an issue in codes with sparse fields. Hopefully the smallness of the data makes this issue negligible.